mono/metadata/threads.c

```diff
@@ -2194,8 +2194,14 @@ void mono_thread_current_check_pending_interrupt ()
 void
 ves_icall_System_Threading_Thread_Abort (MonoInternalThread *thread, MonoObject *state)
 {
+	MonoThread *sys_thread = mono_thread_current ();
+
 	LOCK_THREAD (thread);
 	
+	if (sys_thread->pending_exception) {
```

Comment:
I don't understand why we need that. IMO if have the pending exception, we should still throw the ThreadAbortException, after the pending exception have been catched. The following test should expose the expected behaviour:

```

using System;
using System.Threading;

class CustomException : Exception
{
}

class Driver
{
	public static void Main ()
	{
		try {
			try {
				try {
					try {
						Thread.CurrentThread.Abort ();
					} catch (ThreadAbortException tae0) {
						throw new CustomException ();
					}
				} catch (CustomException ce1) {
					/* ThreadAbortException should be rethrown */
				}
				Console.WriteLine ("failure");
			} catch (ThreadAbortException tae2) {
				Console.WriteLine ("success");
			}
		} catch (Exception e3) {
			Console.WriteLine ("Exception: {0}", e3);
		}
	}
}
```

What I suspect is that this test will fail with this change.